frontend/packages/dev-console/src/components/pipelines/modals/common/PipelineSecretSection.tsx

```diff
@@ -1,66 +1,50 @@
 import * as React from 'react';
-import { Formik, useField, useFormikContext, FormikValues } from 'formik';
+import { useFormikContext, FormikValues } from 'formik';
 import { PlusCircleIcon } from '@patternfly/react-icons';
 import { Button } from '@patternfly/react-core';
 import { ExpandCollapse } from '@console/internal/components/utils';
-import { SecretType } from '@console/internal/components/secrets/create-secret';
 import { SecretModel } from '@console/internal/models';
-import { k8sCreate } from '@console/internal/module/k8s';
-import {
-  associateServiceAccountToSecret,
-  getSecretAnnotations,
-} from '../../../../utils/pipeline-utils';
-import { SecretAnnotationId } from '../../const';
-import { advancedSectionValidationSchema } from './validation-utils';
+import { SecretKind } from '@console/internal/module/k8s';
+import { getSecretAnnotations } from '../../../../utils/pipeline-utils';
 import SecretForm from './SecretForm';
 import SecretsList from './SecretsList';
+import { secretFormDefaultValues } from './utils';
 
 import './PipelineSecretSection.scss';
 
-const initialValues = {
-  secretName: '',
-  annotations: { key: SecretAnnotationId.Image, value: '' },
-  type: SecretType.dockerconfigjson,
-  formData: {},
-};
-
 type PipelineSecretSectionProps = {
   namespace: string;
 };
 
 const PipelineSecretSection: React.FC<PipelineSecretSectionProps> = ({ namespace }) => {
-  const [secretOpenField] = useField<boolean>('secretOpen');
-  const { setFieldValue } = useFormikContext<FormikValues>();
+  const {
+    setFieldValue,
+    values: { newSecrets, secretForm },
+    setFieldTouched,
+  } = useFormikContext<FormikValues>();
 
-  const handleSubmit = (values, actions) => {
-    actions.setSubmitting(true);
+  const handleSubmit = () => {
+    setFieldValue(`secretForm`, secretFormDefaultValues);
+    setFieldTouched(`secretForm`, false);
     const newSecret = {
```

Comment:
```suggestion
    const newSecret: SecretKind = {
```
This will need to update the type, it seems to expect `data` whereas the documentation ([docs](https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret-manually) | [reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#secret-v1-core)) does not indicate all secrets need to have a `data` field. I think we need to [update the type](https://github.com/openshift/console/blob/master/frontend/public/module/k8s/types.ts#L923) to make `data` optional.

cc @christianvogt